www/_template/reference/configuration.md

```diff
@@ -5,110 +5,325 @@ description: The Snowpack configuration API reference.
+
+```js
+// snowpack.config.js
+// Example: alias types
+{
+  alias: {
+    // Type 1: Package Import Alias
+    "lodash": "lodash-es",
+    "react": "preact/compat",
+    // Type 2: Local Directory Import Alias (relative to cwd)
+    "components": "./src/components"
+    "@app": "./src"
+  }
+}
+```
+
+## config.plugins
+
+**Type**: `array` containing pluginName `string` or an array [`pluginName`, {`pluginOptions`}
+
+Enable Snowpack plugins and their options
+
+```js
+// snowpack-config.js
+// Example: enable plugins both simple and expanded
+{
+  plugins: ['plugin-1', ['plugin-2', { 'plugin-option': false }]];
+}
+```
+
+## config.devOptions
+
+**Type**: `object` (option name: value)
+
+Configure the Snowpack dev server.
+
+### devOptions.secure
+
+**Type**: `boolean`
+
+**Default**: `false`
+
+Toggles whether Snowpack dev server should use HTTPS with HTTP2 enabled.
+
+### devOptions.hostname
+
+**Type**: `string`
+
+**Default**: `localhost`
+
+The hostname the dev server runs on.
+
+TODO: originally "The hostname where the browser tab will be open." is this more correct? Also why isn't it in the devoptions object in config.ts?
+
+### devOptions.port
+
+**Type**: `number`
+
+**Default**: `8080`
+
+The port the dev server runs on.
+
+### devOptions.fallback
+
+**Type**: `string`
+
+**Default**: `"index.html"`
+
+The HTML file to serve for non-resource routes.
+
+When using the Single-Page Application (SPA) pattern, this is the HTML "shell" file that gets served for every (non-resource) user route.
+
+⚠️ Make sure that you configure your production servers to serve this.
+
+### devOptions.bundle
+
+**Type**: `boolean`
+
+**Default**: TODO
+
+TODO: This is still in config.ts but seems deprecated
+
+### devOptions.open
+
+**Type**: `string`
+
+**Default**: `"**Default**"`
+
+Configures how the dev server opens in the browser when it starts.
+
+Any installed browser, e.g., "chrome", "firefox", "brave". Set "none" to disable.
+
+### devOptions.output
+
+**Type**: `"stream" | "dashboard"`
+
+**Default**: `"dashboard"`
+
+Set the output mode of the `dev` console:
+
+- `"dashboard"` delivers an organized layout of console output and the logs of any connected tools. This is recommended for most users and results in the best logging experience.
+- `"stream"` is useful when Snowpack is run in parallel with other commands, where clearing the shell would clear important output of other commands running in the same shell.
+
+### devOptions.hmr
+
+**Type**: `boolean`
+
+**Default**: `true`
+
+Toggles HMR on the Snowpack dev server.
+
+### devOptions.hmrDelay
+
+**Type**: `number` (milliseconds)
+
+**Default**: `0`
+
+Milliseconds to delay HMR-triggered browser update.
+
+### devoptions.hmrPort
+
+**Type**: `number`
+
+**Default**: `undefined`
+
+The port where Snowpack's HMR Websocket runs.
+
+The **Default** of `undefined` means it runs on the same port as the Snowpack dev server. TODO I think
```

Comment:
@FredKSchott lmk if this seems correct since I wrote it from scratch

Reply:
haha this was basically right! Cleaned up a bit

Reply:
also it looks like a find and replace got more "Defaults" then you meant. Be careful doing find and replaces like this!